### PR TITLE
feat: Add `GenerateCodeTool` component

### PR DESCRIPTION
**Pull request by rSnapkoOpenOps**
Fixes OPS-2259.

## Testing Checklist

Check all that apply:

- [ ] I tested the feature thoroughly, including edge cases

- [ ] I verified all affected areas still work as expected

- [ ] Automated tests were added/updated if necessary

- [ ] Changes are backwards compatible with any existing data, otherwise a migration script is provided

## Visual Changes (if applicable)

If there are UI/UX changes, include at least one of the following:

- Screenshots
- Loom video
- Preview deployment link

### CODE REVIEW

File: packages/ui-components/src/components/assistant-ui/generate-code-tool/generate-code-tool.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import { Theme } from '@/lib/theme';
```
[the remark this answers is not in the document]
Comment (rmn-snpk) [reply packages/ui-components/src/components/assistant-ui/generate-code-tool/generate-code-tool.tsx]: done

**Comment by alexandrudanpop** [comment]
Need to pass `additionalContext` param to the BE as before:
image

**Comment by alexandrudanpop** [comment]
Do we need this tool container? Ideally for those cases we would not show the tool container:
image

File: packages/ui-components/src/components/assistant-ui/generate-code-tool/generate-code-tool.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+};
+
+const GenerateCodeTool = ({ result, status, theme }: GenerateCodeToolProps) => {
+  const { handleInject } = useThreadExtraContext();
+
+  if (status.type !== 'complete') {
+    return (
+      <Skeleton className="w-full h-[150px]">
+        <div className="flex items-center justify-center h-full">
+          {t('Generating...')}
```
Comment (cezudas) [inline packages/ui-components/src/components/assistant-ui/generate-code-tool/generate-code-tool.tsx]:
```suggestion
          {t('Generating ...')}
```
